src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/AvoidNestedBlocksCheck.java

```diff
@@ -40,7 +40,7 @@
  * {
  *   int whichIsWhich = 0;
  *   {
- *     int whichIsWhich = 2;
```

Comment:
Please revert. Without introduction a new variable, the output is obvious.

Reply:
I'm ok to revert. But before doing that, I'd like to know what is the added value of this check if there is a compilation error?

Reply:
@pbludov ,
I do not understand request. I see change as correct. Example is not real code that is why it is weird. It is ok to change same variable name in inner block, `int ` is a mistake.
Please review one more time.

Reply:
@mincong-h you are right. Java does not allow to "override" variables in the inner scope, as c++ does. But this means that the example is bad. There is no need to guess anything.
But this is not related to this issue.